File: ticket-service/src/main/java/com/helpcore/ticket_service/entidades/dto/MensajeDTO.java
package com.helpcore.ticket_service.entidades.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MensajeDTO {
    private Integer id;
    private Integer idTicket;
    private Integer idUsuario;
    private String tipoUsuario; // CLIENTE o AGENTE
    private String nombreUsuario;
    private String correoUsuario;
    private String mensaje;
    private LocalDateTime fechaCreacion;
    private boolean esDelUsuarioActual;
    private List<ArchivoDTO> archivos;
}

File: helpcore-fe/src/app/dto/mensaje-dto.ts
export interface MensajeDTO {
  id: number;
  idTicket: number;
  idUsuario: number;
  tipoUsuario: string;
  nombreUsuario: string;
  correoUsuario: string;
  mensaje: string;
  fechaCreacion: string;
  esDelUsuarioActual: boolean;
  archivos: ArchivoDTO[];
}

export interface ArchivoDTO {
  id: number;
  nombreOriginal: string;
  nombreAlmacenado: string;
  rutaArchivo: string;
  tipoMime: string;
  tamano: number;
  esImagen: boolean;
  fechaCreacion: string;
}

export interface ChatTicketDTO {
  idTicket: number;
  estadoTicket: string;
  ticketCerrado: boolean;
  mensajes: MensajeDTO[];
  usuarioPuedeMensajear: boolean;
}

export interface RespuestaCrearMensajeDTO {
  success: boolean;
  mensaje: string;
  mensajeCreado: MensajeDTO;
}

export interface RespuestaCargarArchivoDTO {
  success: boolean;
  mensaje: string;
  archivoCreado: ArchivoDTO;
  error?: string;
}
